File: 2022-23/contest2/prog/junior/binary_counting.cpp
#include <bits/stdc++.h>
using namespace std;

string to_string(int num, int base, int width = INT_MAX) {
	string res;
	int i = 0;
	while(num || width != INT_MAX) {
		int d = num % base;
		res += (d + '0');
		num /= base;
		if(++i >= width)
			break;
	}
	reverse(res.begin(), res.end());
	return res;
}

int main() {
	for(int k = 0; k < 5; ++k) {
		string line;
		getline(cin, line);
		string bs;
		for(char ch : line) {
			auto cs = to_string(ch, 2, 8);
			bs += cs;
		}
		int i = 0;
		for(;;++i) {
			auto s = to_string(i, 2);
			if(bs.find(s) == string::npos)
				break;
		}
		cout << i - 1 << endl;
	}
	return 0;
}

File: 2022-23/contest2/prog/junior/binary_counting.py
import os
import sys

INT_MAX = 65536

def to_string(num, base, width = INT_MAX):
	res = ''
	i = 0
	while num or width != INT_MAX:
		d = num % base
		res += str(d)
		num = num // base
		i += 1
		if i >= width:
			break
	return res[::-1] # Reverse a string efficiently in Python

for k in range(5):
	line = input()
	bs = ''
	for ch in line:
		cs = to_string(ord(ch), 2, 8)
		bs += cs
	i = 0
	while(True):
		s = to_string(i, 2)
		if s not in bs:
			break
		i += 1
	print(i - 1)
